boa/src/builtins/value/hash.rs

```diff
@@ -0,0 +1,53 @@
+use super::*;
+
+use crate::builtins::Number;
+use std::hash::{Hash, Hasher};
+
+impl PartialEq for Value {
+    fn eq(&self, other: &Self) -> bool {
+        same_value_zero(self, other)
+    }
+}
+
+impl Eq for Value {}
```

Comment:
This could be derived, right?

Reply:
Sadly it can't #[derive(Eq)] requires that its fields have implemented Eq and f64 does not. (`ValueData::Number(f64)`)